tests/test_variable_manipulation.py

```diff
@@ -68,42 +68,95 @@ def test_find_or_check_numerical_variables(df_vartypes, df_numeric_columns):
+):
     vars_cat = ["Name", "City"]
     vars_mix = ["Age", "Marks", "Name"]
 
-    assert _find_or_check_categorical_variables(df_vartypes, vars_cat) == vars_cat
-    assert _find_or_check_categorical_variables(df_vartypes, None) == vars_cat
-
+    # errors when vars entered by user are not categorical
     with pytest.raises(TypeError):
         assert _find_or_check_categorical_variables(df_vartypes, "Marks")
-
+    with pytest.raises(TypeError):
+        assert _find_or_check_categorical_variables(df_datetime, "datetime_range")
+    with pytest.raises(TypeError):
+        assert _find_or_check_categorical_variables(df_datetime, ["datetime_range"])
     with pytest.raises(TypeError):
         assert _find_or_check_categorical_variables(df_numeric_columns, 3)
-
     with pytest.raises(TypeError):
         assert _find_or_check_categorical_variables(df_numeric_columns, [0, 2])
-
     with pytest.raises(TypeError):
         assert _find_or_check_categorical_variables(df_vartypes, vars_mix)
 
+    # error when user enters empty list
     with pytest.raises(ValueError):
         assert _find_or_check_categorical_variables(df_vartypes, variables=[])
 
+    # error when df has no categorical variables
     with pytest.raises(ValueError):
         assert _find_or_check_categorical_variables(df_vartypes[["Age", "Marks"]], None)
+    with pytest.raises(ValueError):
+        assert _find_or_check_categorical_variables(
+            df_datetime[["date_obj1", "time_obj"]], None
+        )
+
+    # when variables=None
+    assert _find_or_check_categorical_variables(df_vartypes, None) == vars_cat
+    assert _find_or_check_categorical_variables(df_datetime, None) == ["Name"]
 
+    # when vars are specified
+    assert _find_or_check_categorical_variables(df_vartypes, "Name") == ["Name"]
+    assert _find_or_check_categorical_variables(df_datetime, "date_obj1") == [
+        "date_obj1"
+    ]
+    assert _find_or_check_categorical_variables(df_vartypes, vars_cat) == vars_cat
+    assert _find_or_check_categorical_variables(df_datetime, ["Name", "date_obj1"]) == [
+        "Name",
+        "date_obj1",
+    ]
+
+    # vars specified, column name is integer
     assert _find_or_check_categorical_variables(df_numeric_columns, [0, 1]) == [0, 1]
     assert _find_or_check_categorical_variables(df_numeric_columns, 0) == [0]
     assert _find_or_check_categorical_variables(df_numeric_columns, 1) == [1]
 
+    # numeric var cast as category
     df_vartypes["Age"] = df_vartypes["Age"].astype("category")
+    assert _find_or_check_categorical_variables(df_vartypes, "Age") == ["Age"]
     assert _find_or_check_categorical_variables(df_vartypes, None) == vars_cat + ["Age"]
     assert _find_or_check_categorical_variables(df_vartypes, ["Name", "Age"]) == [
         "Name",
         "Age",
     ]
 
+    # datetime vars cast as category
+    # object-like datetime
+    df_datetime["date_obj1"] = df_datetime["date_obj1"].astype("category")
```

Comment:
here we are testing the new functionality. If a variable of time category can be parsed as datetime, then the selector excludes it.

Could we add 1 test to confirm that if the user passes the datetime variable in the variables argument, it is indeed selected?

`assert _find_or_check_categorical_variables(df_datetime, ["Name", "date_obj1"]) == ["Name", "date_obj1"]`